Cmdr/BuiltInCommands/help.lua

```diff
@@ -29,9 +29,27 @@ return {
 			end
 		else
 			local commands = context.Cmdr.Registry:GetCommands()
-
-			for _, cmd in pairs(commands) do
-				context:Reply(("%s - %s"):format(cmd.Name, cmd.Description))
+			local groups = context.Cmdr.Registry:GetGroups()
+			
+			context:Reply([[
+Notes to consider!
+------------------
+. = Me/Self
+* = All/Everyone
+** = Others
+? = Random
+?N = List of N random values	
+]])
+			for _, groupName in pairs(groups) do
+				context:Reply(([[
+%s
+------------------
+]]):format(groupName))
+				for _, cmd in pairs(commands) do
+					if cmd.Group == groupName then
+						context:Reply(("%s - %s"):format(cmd.Name, cmd.Description))
+					end
+				end
```

Comment:
Having a nested for loop here is unnecessary and means we have to loop over *every* command for each group.

Instead, we should get the list of commands, sort the table by their group, and then loop over the list once. When the group changes from the previous group, insert a heading line with the group name. We also don't need to use the multi line literal syntax here.